### ISSUE TITLE
[scan] Test crashed with signal kill before starting test execution

### ISSUE THREAD

**Issue by testableapple** [fastlane/fastlane#21025]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

Sometimes, UI tests crash on startup on CI (see [this job run](https://github.com/GetStream/stream-chat-swift/actions/runs/4072855107/jobs/7022695765#step:4:1204) as an example)

##### Command executed

```ruby
  scan_options = {
    project: xcode_project,
    scheme: scheme,
    testplan: scheme,
    result_bundle: true,
    devices: device,
    number_of_retries: 3
  }
  scan(
    scan_options.merge(
      clean: true,
      build_for_testing: true
    )
  )
  scan(
    scan_options.merge(
      test_without_building: true,
      derived_data_path: lane_context[SharedValues::SCAN_DERIVED_DATA_PATH],
      only_testing: only_testing_batch
    )
  )
```

##### Complete output when running fastlane, including the stack trace and command used

```bash
2023-02-02 15:23:58.171 xcodebuild[16883:44216] [MT] IDETestOperationsObserverDebug: 462.731 elapsed -- Testing started completed.
2023-02-02 15:23:58.171 xcodebuild[16883:44216] [MT] IDETestOperationsObserverDebug: 0.000 sec, +0.000 sec -- start
2023-02-02 15:23:58.171 xcodebuild[16883:44216] [MT] IDETestOperationsObserverDebug: 462.731 sec, +462.731 sec -- end
Testing failed:
	StreamChatUITestsAppUITests-Runner (19377) encountered an error (Early unexpected exit, operation never finished bootstrapping - no restart will be attempted. (Underlying Error: Test crashed with signal kill before starting test execution. If you believe this error represents a bug, please attach the result bundle at /Users/runner/work/stream-chat-swift/stream-chat-swift/fastlane/test_output/StreamChatUITestsApp.xcresult))

** TEST EXECUTE FAILED **
[15:24:07]: Exit status: 65
```

### Environment

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.211.0 | ✅ Up-To-Date  |

**Comment by alteral** [comment]
Hi @TomaszLizer, seems like the ticket you linked reports the error when `fastlane` exits with the signal `abrt`. When this ticket is about signal `kill`. I don't think they are related to each other.

Regarding the original issue I created, I don't think it's fastlane related either, it's more like a CI service performance issue (Github Actions in my case) when Simulator is stuck during `simctl boot` command execution. I've opened a [PR](https://github.com/fastlane/fastlane/pull/21026) to prevent this from the `fastlane` side.

After I implemented the execution of `simctl bootstatus` command on CI straight before starting `fastlane scan`, this error did not pop up anymore.

**Comment by TomaszLizer** [comment]
@alteral In the end I think root cause was similar - we are using github actions too and prelaunching simulator fixed issue on our side. Also that wasn't reproducible locally nor on simpler projects.
Hope your PR gets merged soon so that no one else faces it - such kind of "non reproducible" issues are the worst 😬

**Comment by alteral** [comment]
@TomaszLizer, nice! Really glad that pre-booting a simulator helped you too 🙂

> Hope your PR gets merged soon so that no one else faces it - such kind of "non reproducible" issues are the worst 😬

Yeah, 100%! Hopefully fastlane core team will be able to take a look at my PR someday.
